### PR TITLE
Add StatusCode enum

### PR DESCRIPTION
**Pull request by sethmlarson**
Pulled from the official registry. I made this one a `.py` file because it doesn't actually use any type information.

### CODE REVIEW

**Comment by pquentin** [comment]
Thanks!

File: src/hip/status_codes.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    PROXY_AUTHENTICATION_REQUIRED = 407
+    REQUEST_TIMEOUT = 408
+    CONFLICT = 409
+    GONE = 410
+    LENGTH_REQUIRED = 411
+    PRECONDITION_FAILED = 412
+    PAYLOAD_TOO_LARGE = 413
+    URI_TOO_LONG = 414
+    UNSUPPORTED_MEDIA_TYPE = 415
+    REQUESTED_RANGE_NO_SATISFIABLE = 416
```
[the remark this answers is not in the document]
Comment (sethmlarson) [reply src/hip/status_codes.py]: Nice catch, wanna open a PR to `api-design`? :)

[the remark this answers is not in the document]
Comment (jab) [reply src/hip/status_codes.py]: Sure, can do later this afternoon:)

[the remark this answers is not in the document]
Comment (jab) [reply src/hip/status_codes.py]: Just sent you #161. Thanks for waiting and for the chance to contribute! 😊

**Pull request merged**